tools/android/packaging/xbmc/src/org/xbmc/kodi/Splash.java.in

```diff
@@ -544,8 +544,13 @@ public class Splash extends Activity {
   }
 
   protected void startXBMC() {
-    // NB: We only preload lib@APP_NAME_LC@ to be able to get info on missing symbols.
-    //     This is not normally needed
+    // The dynamic linker on droid is a mess.
+    // Depending on version, it might not be able to properly resolve
+    try
+    {
+      System.loadLibrary("smbclient");
+    }
+    catch (UnsatisfiedLinkError e) {}
```

Comment:
Not really. 3.0 android was static, while everyone uses 3.6 dynamic apparently. The try-catch allows to still run with the static lib.
I shouldn't have to preload the lib, really, but the droid dloader is not behaving on, e.g. aftv.